### PR TITLE
[DROOLS-6284] Rules not fired when RUN_TYPE==STANDARD_FROM_DRL and wh…

### PR DESCRIPTION
**Pull request by tkobayas**
…en the constraint is jitted

Fixed for these 3 tests:
org.drools.modelcompiler.TypeObjectCoercionTest.testJoinStringToObject2[STANDARD_FROM_DRL]
org.drools.modelcompiler.TypeObjectCoercionTest.testJoinObjectToString2[STANDARD_FROM_DRL]
org.drools.modelcompiler.TypeObjectCoercionTest.testJoinStringToObjectNonComparable[STANDARD_FROM_DRL]

Also tested drools-model-compiler and drools-test-coverage with `mvn clean test -Ddrools.jittingThreshold=0` locally.

**JIRA**: 

https://issues.redhat.com/browse/DROOLS-6284
https://issues.redhat.com/browse/RHDM-1595

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: drools-model/drools-model-compiler/src/test/java/org/drools/modelcompiler/TypeObjectCoercionTest.java [hunk 1]
```diff
@@ -212,6 +213,7 @@ public void testJoinStringToObject5() {
         ksession.dispose();
     }
 
+    @Ignore("in case of standard-drl, MathProcessor.doOperationNonNumeric() returns false when the left operand is not Comparable. But it doesn't make sense to keep compatibility for this behaviour")
```
Comment (hellowdan) [inline drools-model/drools-model-compiler/src/test/java/org/drools/modelcompiler/TypeObjectCoercionTest.java]: Hi Toshiya, thanks for the PR! Would it be possible to add a test (or change this one) that somehow expects NumberFormatException for this scenario instead of ignoring it?

**Review by hellowdan — APPROVED** [review]

**Review by mariofusco — APPROVED** [review]

File: drools-model/drools-model-compiler/src/test/java/org/drools/modelcompiler/TypeObjectCoercionTest.java (hunk 1 above)
Comment (mariofusco) [reply drools-model/drools-model-compiler/src/test/java/org/drools/modelcompiler/TypeObjectCoercionTest.java]: @tkobayas I agree with @hellowdan and in general I think that it doesn't make sense to keep ignored tests in our codebase if we know in advance that they are structurally broken or anyway we can't or won't to fix them. Please change this test according to Daniel's suggestion or if not possible just remove it.

Comment (tkobayas) [reply drools-model/drools-model-compiler/src/test/java/org/drools/modelcompiler/TypeObjectCoercionTest.java]: Hi @mariofusco @hellowdan , It means we will change Mvel to throw an Exception (probably RuntimeException or IllegalArgumentException rather than NumberFormatException because it's not only for number) when a left operand is not Comparable? It's already an irregular constraint but the change would have a wide effect.

https://github.com/mvel/mvel/blob/master/src/main/java/org/mvel2/math/MathProcessor.java#L269-L279

And then other engines (Jitted standard-drl and exec-model) throw the same Exception for compatibility.  Am I understanding correctly? I think it's not impossible but we may not want to do it for such a minor use case so shall I remove the test? Please confirm, thanks!

Comment (mariofusco) [reply drools-model/drools-model-compiler/src/test/java/org/drools/modelcompiler/TypeObjectCoercionTest.java]: @tkobayas that's too much effort and there are too many corner cases imo. If my understanding is correct and this test is referring to a wrong or misleading behaviour that we don't want to support, please just totally remove the test.

Comment (hellowdan) [reply drools-model/drools-model-compiler/src/test/java/org/drools/modelcompiler/TypeObjectCoercionTest.java]: I agree

Comment (tkobayas) [reply drools-model/drools-model-compiler/src/test/java/org/drools/modelcompiler/TypeObjectCoercionTest.java]: Okay, removed.
